sphinx/source/developers/threading.rst

```diff
@@ -0,0 +1,30 @@
+Threading
+=========
+
```

Comment:
I think we should add a blurb describing the threading in general terms. It gives us performance multipliers since transactions can be processed in parallel, and moves client work to worker threads where they shouldn't be able to block  service/consensus progress (ie #557).